### PR TITLE
fix: GH-374 Default alias name when creating a new alias

### LINKED ISSUE #374: alias add -f directory should create jbang-catalog.json in that directory
would be nice to just go:

`jbang alias -f . xyz path/to/xyz.java`

instead of having to spell out jbang-catalog.json.

right now the above gives: `[jbang] [WARN] Unable to add alias: /Users/max/code/camel-visual/.: Is a directory`

### PR DESCRIPTION
**Pull request by chrsblck**
Fixes #374

### CODE REVIEW

**Comment by chrsblck** [comment]
Shoot, reading through the `Semantic PR` docs, sorry :(

**Comment by maxandersen** [comment]
fix looks right - can you add a test to verify it so we don't accidentally break it ? :) 

`itests/catalog.feature` or `src/test/java/dev/jbang/cli/TestCatalog.java` would be options for places to add that.

**Comment by chrsblck** [comment]
Actually, there's a bug here, I think. I'll add 1 or 2 tests, good call!

**Comment by chrsblck** [comment]
Yeah, I did introduce a bug by not resolving the default path before fetching any existing aliases. I've fixed this and added unit tests to cover this case and the case where the file doesn't exist.

That said, do you think this is the right test file (`TestCatalog.java`)?

**Review by quintesse — CHANGES_REQUESTED** [review]
Ok, so although this is indeed correct, I'd prefer it a bit differently.

First of all you've now only solved it for `alias add`, but the other alias commands also support the `-f` option which means that now `alias add` has a feature the other commands don't have. So my first suggestions is to make it work for all of them.

And second, you solved this inside the API, which although perfectly correct, is not where I'd put this. I'd rather have that the API doesn't have any "tricks" to make things easier for the user, but that this would be handled by the code that "interfaces" with the user. In this case the commands in the `cli` package.

And in fact, there is a perfect spot for doing this: https://github.com/jbangdev/jbang/blob/master/src/main/java/dev/jbang/cli/Alias.java#L32 , which is where the `-f` option is handled for all alias commands.

**Comment by chrsblck** [comment]
@quintesse Makes sense to me, thanks for the review. I'll fix this up tonight.

**Comment by chrsblck** [comment]
@quintesse or @maxandersen should I move the tests? Possibly to `TestAlias.java`

**Comment by quintesse** [comment]
Yeah, that's probably a good idea. TestAlias is for tests related to the alias command. (Although honestly it's all a bit mixed)

**Comment by chrsblck** [comment]
> Yeah, that's probably a good idea. TestAlias is for tests related to the alias command. (Although honestly it's all a bit mixed)

Yeah, it wasn't totally clear to me if that was the right thing to do. Thanks!

**Comment by chrsblck** [comment]
@quintesse Fixed. Ready for another review when you have a chance.

I'll squash the changes once you sign off 😄

File: src/main/java/dev/jbang/cli/Alias.java [hunk 1]
```diff
@@ -29,6 +31,9 @@ protected Path getCatalog() {
 		if (global) {
 			return Settings.getAliasesFile();
 		} else {
+			if (Files.isDirectory(catalogFile)) {
+				catalogFile = Paths.get(catalogFile.toString(), AliasUtil.JBANG_CATALOG_JSON);
```
Comment (maxandersen) [inline src/main/java/dev/jbang/cli/Alias.java:35]: should it not just return the `Paths.get(...)` result and not change the `catalogFile` ?

Comment (quintesse) [reply src/main/java/dev/jbang/cli/Alias.java:35]: Indeed :-)

Comment (chrsblck) [reply src/main/java/dev/jbang/cli/Alias.java]: Fixed :)

**Review by quintesse — APPROVED** [review]
LGTM

**Comment by maxandersen** [comment]
Thanks @chrsblck !

@all-contributors please add @chrsblck for code

**Comment by maxandersen** [comment]
oh missed this - but for next time GH-374 is jira style issue reference, in github just use #374 and best to just have it in body of commit message as otherwise linking doesn't work that great.

**Comment by chrsblck** [comment]
OK, will do. (yes, we use jira + github at work, so looked right to me 😄 )

**Comment by maxandersen** [comment]
ha, turns out GH-374 works too...did not know that - I guess github got tired of jira syntax failing ;)

**Comment by chrsblck** [comment]
Haha, yeah, I saw it in their docs somewhere... Or maybe not, I could have seen it here https://stackoverflow.com/a/6742691/1246282.

Either way, I won't use that for non jira tagging. Thanks for the pointer!

**Pull request merged**